system/jlib/jlog.hpp

```diff
@@ -1342,4 +1348,301 @@ interface IComponentLogFileCreator : extends IInterface
+        offset = l.offset;
+
+        return *this;
+    }
+
+    ~LogAccessConditions()
+    {
+        logFieldNames.kill(true);
+    }
+
+    void appendLogFieldName(const char * fieldname)
+    {
+        logFieldNames.append(fieldname);
+    }
+
+    void copyLogFieldNames(const StringArray * fields)
+    {
+        if (*fields)
+        {
+            ForEachItemIn(fieldsindex,*fields)
+            {
+                logFieldNames.append(fields->item(fieldsindex));
+            }
+        }
+    }
+    inline const StringArray &queryLogFieldNames() { return logFieldNames; }
+};
+
+interface IRemoteLogAccess : extends IInterface
+{
+    virtual bool fetchLog(LogAccessConditions & options, StringBuffer & returnbuf) = 0;
+
+    virtual const char * getRemoteLogAccessType() = 0;
+    virtual IPropertyTree * fetchLogMap() = 0;
+    virtual const char * fetchConnectionStr() = 0;
+};
+
+inline bool fetchLog(StringBuffer & returnbuf, IRemoteLogAccess & logAccess, ILogAccessFilter * filter, LogAccessTimeRange timeRange, StringArray & cols)
+{
+    LogAccessConditions logFetchoptions;
+    logFetchoptions.timeRange = timeRange;
+    logFetchoptions.filter = filter;
+    logFetchoptions.copyLogFieldNames(&cols); //ensure these fields are declared in m_logMapping->queryProp("WorkUnits/@contentcolumn")? or in LogMap/Fields?"
+
+    return logAccess.fetchLog(logFetchoptions, returnbuf);
+}
+
+inline bool fetchJobIDLog(StringBuffer & returnbuf, IRemoteLogAccess & logAccess, const char *jobid, LogAccessTimeRange timeRange, StringArray & cols)
+{
+    return fetchLog(returnbuf, logAccess, getWUIDLogAccessFilter(jobid), timeRange, cols);
+}
+
+inline bool fetchComponentLog(StringBuffer & returnbuf, IRemoteLogAccess & logAccess, const char * component, LogAccessTimeRange timeRange, StringArray & cols)
+{
+    return fetchLog(returnbuf, logAccess, getComponentLogAccessFilter(component), timeRange, cols);
+}
+
+inline bool fetchLogByAudience(StringBuffer & returnbuf, IRemoteLogAccess & logAccess, MessageAudience audience, LogAccessTimeRange timeRange, StringArray & cols)
+{
+    return fetchLog(returnbuf, logAccess, getAudienceLogAccessFilter(audience), timeRange, cols);
+}
+
+inline bool fetchLogByClass(StringBuffer & returnbuf, IRemoteLogAccess & logAccess, LogMsgClass logclass, LogAccessTimeRange timeRange, StringArray & cols)
+{
+    return fetchLog(returnbuf, logAccess, getClassLogAccessFilter(logclass), timeRange, cols);
+}
+
+//logAccessPluginConfig expected to contain connectivity and log mapping information
+typedef IRemoteLogAccess * (*newLogAccessPluginMethod_t_)(IPropertyTree & logAccessPluginConfig);
+
+class jlib_decl RemoteLogAccessLoader
+{
+public:
+    RemoteLogAccessLoader() = default;
+    ~RemoteLogAccessLoader();
+
+    template <class RemoteLogAccessPlugin>
+    static RemoteLogAccessPlugin * queryRemoteLogAccessor()
+    {
+        Owned<IPropertyTree> logAccessPluginConfig = getGlobalConfig()->getPropTree("logaccess");
```

Comment:
this leaks (getGlobalConfig in particular), you either need to own (Owned it) or release it some way, or you can use getGlobalConfigSP() in this context.